src/registered.rs

```diff
@@ -65,30 +127,39 @@ impl SecretKey {
     /// Derives a key for a registered application protocol at the given path from the
     /// given seed. Each path element may consist of an index and (possibly empty) tag.
     ///
-    /// - `zip_number` is the number of the ZIP defining the application protocol.
-    ///   The corresponding hardened index (with empty tag) will be prepended to the
-    ///   `subpath` to obtain the full derivation path.
-    /// - `context_string` is an identifier for the context in which this key will be
-    ///   used. It must be globally unique.
-    ///
-    /// # Panics
-    ///
-    /// Panics if:
-    /// - the context string is empty or longer than 252 bytes.
-    /// - the seed is shorter than 32 bytes or longer than 252 bytes.
+    /// - `context_string`: an identifier for the context in which this key will be used. It must
+    ///   be globally unique, non-empty, and no more than 252 bytes in length.
+    /// - `seed`: the root seed. Must be between 32 bytes and 252 bytes in length, inclusive.
+    /// - `zip_number`: the number of the ZIP defining the application protocol. The corresponding
+    ///   hardened index (with empty tag) will be prepended to the `subpath` to obtain the ZIP 32
+    ///   path.
+    /// - `subpath`: the path to the desired child element. A non-empty path is required, in order
+    ///   to ensure that the resulting full-width cryptovalue is within the allowed subtree rooted
+    ///   at `m_{context} / zip_number'`.
```

Comment:
```suggestion
    /// - `subpath`: the path to the desired child element.
```
I was mistaken here; an empty path was intentionally allowed for this API. There is no problem with it because it is within the allowed subtree.